decision-records/2024-07-16--shared-urls.md

```diff
@@ -0,0 +1,61 @@
+# Shared URLs between COS-Lite components and external charms
+**Date:** 2024-07-16<br/>
+**Authors:** @Abuelodelanada, @sed-i
+
+
+## Context and Problem Statement
+The workload of many charms is a server, that other charmed workloads need to reach.
+Typically, the server URL is "announced" over relation data, but we have different URLs "types" we could announce:
+```
+{HTTP / HTTPS} {internal / ingress} {IP / FQDN}
```

Comment:
Can't we express these combinations in another way? From my point of view it is complex to understand at glance. What do you think about this??

```mermaid
flowchart LR
    A["HTTP | HTTPS"] --> B["internal IP | internal FQDN"]
    A --> C["ingress IP | ingress FQDN"]
```

Reply:
Made it more explicit in next commit.